src/include/printer/connectionwizard.rb

```diff
@@ -288,56 +288,27 @@ def getUriWithUsernameAndPassword(uri, scheme)
           "."
         )
         # Let the whole pipe fail if any of its commands fail (requires bash):
-        grepcommand = Ops.add(
-          Ops.add(
-            Ops.add(
-              Ops.add(
-                Ops.add("set -o pipefail ; egrep '^DeviceURI ", scheme),
-                "://[^:]+:[^@]+@"
-              ),
-              part1
-            ),
-            "/"
-          ),
-          part2
-        )
+        grepcommand = "set -o pipefail ; egrep '^DeviceURI " + scheme + "://[^:]+:[^@]+@" + part1 + "/" + part2
```

Comment:
scheme, part[12] need to be escaped. Some characters are handled for part[12] with special_chars, but not all

Reply:
Fixed.